tests/test_caching.py

```diff
@@ -70,27 +70,27 @@ def test_compare_cached_vs_uncached(self):
         cached_trace = trappy.FTrace(uncached_trace.trace_path)
         cached_dfr = cached_trace.sched_wakeup.data_frame
 
-        # Test whether timestamps are the same:
-        # The cached/uncached versions of the timestamps are slightly
-        # different due to floating point precision errors due to converting
-        # back and forth CSV and DataFrame. For all purposes this is not relevant
-        # since such rounding doesn't effect the end result.
-        # Here's an example of the error, the actual normalized time when
-        # calculated by hand is 0.081489, however following is what's stored
-        # in the CSV for sched_wakeup events in this trace.
-        # When converting the index to strings (and also what's in the CSV)
-        # cached: ['0.0814890000001', '1.981491']
-        # uncached: ['0.0814890000001', '1.981491']
+        # By default, the str to float conversion done when reading from csv is
+        # different from the one used when reading from the trace.txt file.
         #
-        # Keeping index as numpy.float64
-        # cached: [0.081489000000100009, 1.9814909999999999]
-        # uncached: [0.081489000000146916, 1.9814909999995507]
+        # Here's an example:
+        # - trace.txt string timestamps:
+        #   [76.402065, 80.402065, 80.001337]
+        # - parsed dataframe timestamps:
+        #   [76.402065000000007, 80.402065000000007, 82.001337000000007]
         #
-        # To make it possible to test, lets just convert the timestamps to strings
-        # and compare them below.
+        # - csv string timestamps:
+        #   [76.402065, 80.402065, 80.001337]
+        # - cached dataframe timestamps:
+        #   [76.402064999999993, 80.402064999999993, 82.001337000000007]
+        #
+        # To fix this, the timestamps read from the cache are rounded
+        # to microsecond precision, which results in cache-read timestamps
+        # being identical to trace-read timestamps. This test ensures that
+        # this stays true.
```

Comment:
This comment is out of date now

Reply:
I updated it but forgot to add it to the commit, I'll push it right now